client/app/src/main/java/com/example/protego/SignupActivity.java

```diff
@@ -119,12 +119,13 @@ public void onSuccess(AuthResult authResult) {
 
                         Log.d(TAG, "Creating the user...");
 
-                        FirebaseUser firebaseUser = authResult.getUser();
-
+                        FirebaseUser firebaseUser = FirebaseAuth.getInstance().getCurrentUser();
 
                         ProtegoUser protegoUser = new ProtegoUser();
                         protegoUser.setFirstName(first_name_input.getText().toString());
-                        protegoUser.setLastName(last_name_input.getText().toString());
+                        // the user is a doctor so set last name
+                        if (last_name_input != null)
```

Comment:
We'll need a better way to check this if we add a last name field to patient registration.